modules/volunteers/admin/sitesupport/checkout.js

```diff
@@ -0,0 +1,223 @@
+                }
+              });
+          }
+        });
+      });
+    },
+    clearCheckout() {
+      hideSidebar();
+      this.hoursSpent = 0;
+      this.checkout = [];
+      this.groupsNow = [];
+    },
+    updateCost(cost) {
+      this.hoursSpent += cost;
+    },
+    addToCheckout(item) {
+      this.init();
+      var cost = parseFloat(item.value);
+      var found = false;
+      var target = null;
+      this.checkout.every((element) => {
+        if (element.id == item.id) {
+          found = true;
+          target = element;
+          return false;
+        }
+        return true;
+      });
+
+      if (item.promo == '1') {
+        if (found) {
+          alertbox('Promo item \'' + item.name + '\' cannot be added twice');
+          return;
+        }
+        cost = 0;
+      }
+      if (this.$parent.$refs.gfttbl.hoursRemaining < this.hoursSpent + cost) {
+        alertbox('Volunteer does not have enough hours for the ' + item.name);
+        return;
+      }
+      if (item.reward_group !=  null) {
+        if (!(item.reward_group.id in this.groupsNow)) {
+          this.groupsNow[item.reward_group.id] = 0;
+        } else {
+          if (this.groupsNow[item.reward_group.id] + 1 > item.reward_group.reward_limit) {
+            alertbox('Too many items from limited group');
+            return;
+          }
+        }
+      }
+      var count = 1;
+      if (target != null) {
+        count += target.count;
+      }
+
+      if (item.claimed + count > item.inventory) {
+        alertbox('Not enough items in inventory!');
+        return;
+      }
+
+      if (target == null) {
+        item.count = count;
+        this.checkout.push(item);
+      } else {
+        target.count = count;
+      }
+
+      this.updateCost(cost);
+
+      if (item.reward_group !== null) {
+        this.groupsNow[item.reward_group.id] += 1;
+      }
+
+      this.show();
+    },
+    addPromoToCheckout(unclaimed) {
+      unclaimed.forEach((item) => {
+        if (item.reward_group == null) {
+          this.addToCheckout(item);
+        }
+      });
+    },
+    removeFromCheckout(index, item) {
+      this.updateCost(-1 * parseFloat(item.value));
+      if (item.count > 1) {
+        item.count -= 1;
+      } else {
+        this.checkout.splice(index, 1);
+      }
+      if (item.reward_group !== null) {
+        this.groupsNow[item.reward_group.id] -= 1;
+      }
+    },
+    init() {
+      if (this.groupsNow.length == 0) {
+        this.$parent.$refs.gfttbl.groupCount.forEach((entry, index) => {
+          this.groupsNow[index] = entry;
+        });
+      }
+    },
+    show() {
+      this.init();
+      showSidebar('checkout_div');
+    },
+    printHoursLeft() {
+      if (this.$parent.$refs.gfttbl) {
+        return this.$parent.printHours(this.$parent.$refs.gfttbl.hoursRemaining - this.hoursSpent);
+      }
+    },
+    printHoursSpent() {
+      return this.$parent.printHours(this.hoursSpent);
+    }
+  },
+  template: `
```

Comment:
I noticed Vue documentation provides another way to describe templates as well, where it looks more like HTML rather than a string. Is there a reason that we are using this approach that I haven't seen yet in Vue docs?

Reply:
I believe it is called `object-based syntax`, but it is also referred to as non-script setup SFC format.  The basic reason is that I was initial taught Vue using it and it has done everything I need so I have not moved on from there.

Reply:
I see it now, thanks for pointing that out. Looks like this is the approach if we are not doing build-time things to it, so that makes sense.